packages/core/src/state-management/repository.ts

```diff
@@ -487,28 +496,40 @@ export class Repository {
     // We also skip CACAO expiration checking during this initial load as its possible
     // that the CommitID we are being asked to load may in fact be an anchor commit with
     // the timestamp information that will reveal to us that the CACAO didn't actually expire.
-    const base$ = await this.load(commitId.baseID, opts, false)
-
-    return this._atCommit(commitId, base$)
-  }
+    let existingState$ = null
+    try {
+      existingState$ = await this.load(commitId.baseID, opts, false)
+    } catch (err) {
+      this.logger.warn(
+        `Error loading existing state for stream ${commitId.baseID} while loading stream at commit ${commitId.commit}`
+      )
+    }
 
-  private async _atCommit(
-    commitId: CommitID,
-    existingState$: RunningState
-  ): Promise<SnapshotState> {
     return this.executionQ.forStream(commitId).run(async () => {
-      const stateAtCommit = await this.streamLoader.stateAtCommit(existingState$.state, commitId)
+      const stateAtCommit = existingState$
+        ? await this.streamLoader.resetStateToCommit(existingState$.state, commitId)
+        : await this.streamLoader.stateAtCommit(commitId)
 
       // Since we skipped CACAO expiration checking earlier when we loaded the current stream state,
       // we need to make sure to do it here.
       SignatureUtils.checkForCacaoExpiration(stateAtCommit)
 
-      // If the provided CommitID is ahead of what we have in the cache and state store, then we
-      // should update them to include it.
-      if (StreamUtils.isStateSupersetOf(stateAtCommit, existingState$.value)) {
-        existingState$.next(stateAtCommit)
-        await this._updateStateIfPinned(existingState$)
+      // Check if we need to update what state is stored in our state store based on any
+      // information learned from this CommitID.
+      if (existingState$) {
+        if (StreamUtils.isStateSupersetOf(stateAtCommit, existingState$.value)) {
+          // If the provided CommitID is ahead of what we have in the cache and state store, then we
+          // should update them to include it.
+          existingState$.next(stateAtCommit)
```

Comment:
For a case were actual commits are A -> B -> C. what happens here if existing state is at 'A' and commit we learned is at 'C' 
Will this change it to - A -> C ? Or will we load ( B -> C ) and get the correct order of A -> B -> C